src/org/mirah/jvm/mirrors/generics/async_type_builder.mirah

```diff
@@ -37,12 +37,13 @@ interface AsyncTypeBuilderResult
 end
 
 class AsyncTypeBuilder < SignatureVisitor
-  def initialize(context:Context, typeVariables:Map={}):void
+  def initialize(context:Context, typeVariables:Map, processed_signatures:Map):void
```

Comment:
processed_signatures is a little too generic of a name :smile: . How about `seen_signatures`? I think that would make it more clear that it's a collection of signatures we've already created types for but may or may not have been resolved.